llvm/lib/Target/AIE/AIEHazardRecognizer.h

```diff
@@ -179,13 +181,20 @@ class AIEHazardRecognizer : public ScheduleHazardRecognizer {
   /// For efficiency, this size is rounded up to a power of two.
   unsigned computeScoreboardDepth() const;
 
+  bool checkConflict(const ResourceScoreboard<FuncUnitWrapper> &Scoreboard,
+                     const MCInstrDesc &Desc, const int DeltaCycles) const;
```

Comment:
Do we need even more of those wrappers? This one seems somewhat error-prone as it ignores register-based itineraries and memory banks.

Reply:
Agreed. This is a level zero resolution of a conflict between Krishnam's PR and mine. I was hesitant to bring in MachineInstr internals here, but I would really like to have a concise conflict checker with an external scoreboard.

Reply:
In fact, it is dead after that rebase.